### ISSUE TITLE
2FA / MFA

### ISSUE THREAD

**Issue by drwetter** [backuppc/backuppc#409]
Hi,

does backuppc currently support MFA/2FA of any kind? Any plans to add something like [this](https://metacpan.org/pod/Authen::HOTP) (no clue whether this is any good).

Cheers, Dirk

**Comment by grzegorznowak** [comment]
Hey, out of curiosity, what is the use case where you would actually need to have 2fa in front of the BackupPC  @drwetter ?

To me it feels that exposing it beyond `localhost` is going to always be asking for more problems than it can solve.

**Comment by drwetter** [comment]
I was not suggesting to enforce it for all users but supply optionally.

Okay: backups are in general juicy targets: they probably carry confidential data and the way BackupPC operates: It has administrative access to all nodes being backuped. So whoever has full access to BackupPC has also full access to data on the nodes currently and from the past.

These "days" it seems even more recommended to have a second factor: ransomware. If a company backup is encrypted too the ransom and any damage if the date cannot be recovered is higher. If BackupPC's GUI is operated from a Windows PC which may or may not be in a domain, the Windows PC should be treated as owned. A second factor, properly used, could give an additional layer of protection.

Hope that clarifies my request?

**Comment by grzegorznowak** [comment]
I get all that, but as a long-term user of backuppc myself I was simply wondering what setup would warrant 2FA being anything other than just a stub security. I think that in principle you wouldn't want your backupGUI exposed beyond perimeters of the server running the software.

You are perfectly correct with being worried about ransomewarez, that's why having the backuppc detached from everything and hidden in a vault is the key imho. To the point of actively rotating ssh keys used to access the resource via a tunnel or however else you do it. Applying zero trust principles in this context would mean a need to go through a grant process to generate the ssh key that will get you within perimeters of the service. And that suddenly moves the requirement of applying any SecOps from backuppc software onto the wrapping layer of whatever hosts it - so generally a more modular and scalable approach.

Maybe try this route instead 🤔  ?

**Comment by drwetter** [comment]
I raised the issue not because I am using it and thought it would be useful for myself. I raised it bc I believe it would be a good thing in general for professional environments at least.

Folks exposing it to the outside are lost souls and we don't want to argue about that. Ransomware is happening behind the perimeter.

I sense what you're saying but detached from everything doesn't work as you still need this thing to backup everything :-)  The other suggestions may fit for some folks, for others not. And you can be pretty restricted in a corporate environment concerning other suggestions.

In the end somebody's administrating BackupPC with the GUI. And if the desktop of that individual is compromised the second factor puts the bar higher!

**Comment by grzegorznowak** [comment]
Yeah I definitely see a value in having more enterprise'y vibe to the software in general. No argue there. It just feel like the it is currently far far away from this happening, thus using it behind onion fortifications is the next best thing imho.

Connected it has to be, but if it's a read-only connection then the impact is negligible, UNLESS we start thinking about more sophisticated vectors of attack, ie. pulling down spoofed files that explode when being processed by the backuppc logic. I'm too IQ limited to even fathom how that could look like, but one thing I'm sure of is that there are pure evil genius people who can reverse-engineer the system and pull it off.

btw. long live OWASP ;-)

**Comment by ediazrod** [comment]
My 2FA is a .haccess file to avoid external Ips, use the tool. As is Apache autentication sure that you can use apache to autenticate on Backuppc with 2FA.

**Comment by GWHAYWOOD** [comment]
Closing, no plans to implement 2FA at this time.
